connect-inject/handler_test.go

```diff
@@ -1189,6 +1277,146 @@ func TestShouldInject(t *testing.T) {
+					HTTPGet: &corev1.HTTPGetAction{
+						Port: intstr.FromInt(8080),
+					},
+				},
+			},
+			expReadinessPort:         8080,
+			expOriginalReadinessPort: 0,
+		},
+		"transparent proxy disabled; overwrite probes enabled": {
+			tproxyEnabled:   false,
+			overwriteProbes: true,
+			readinessProbe: &corev1.Probe{
+				Handler: corev1.Handler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Port: intstr.FromInt(8080),
+					},
+				},
+			},
+			expReadinessPort:         8080,
+			expOriginalReadinessPort: 0,
+		},
+		"just the readiness probe": {
+			tproxyEnabled:   true,
+			overwriteProbes: true,
+			readinessProbe: &corev1.Probe{
+				Handler: corev1.Handler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Port: intstr.FromInt(8080),
+					},
+				},
+			},
+			expReadinessPort:         defaultExposedPathsListenerPortReadiness,
+			expOriginalReadinessPort: 8080,
+		},
+		"just the liveness probe": {
+			tproxyEnabled:   true,
+			overwriteProbes: true,
+			livenessProbe: &corev1.Probe{
+				Handler: corev1.Handler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Port: intstr.FromInt(8081),
+					},
+				},
+			},
+			expLivenessPort:         defaultExposedPathsListenerPortLiveness,
+			expOriginalLivenessPort: 8081,
+		},
+		"both readiness and liveness probes": {
+			tproxyEnabled:   true,
+			overwriteProbes: true,
+			livenessProbe: &corev1.Probe{
+				Handler: corev1.Handler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Port: intstr.FromInt(8081),
+					},
+				},
+			},
+			readinessProbe: &corev1.Probe{
+				Handler: corev1.Handler{
+					HTTPGet: &corev1.HTTPGetAction{
+						Port: intstr.FromInt(8080),
+					},
+				},
+			},
+			expLivenessPort:          defaultExposedPathsListenerPortLiveness,
+			expOriginalLivenessPort:  8081,
+			expReadinessPort:         defaultExposedPathsListenerPortReadiness,
+			expOriginalReadinessPort: 8080,
+		},
+	}
+
+	for name, c := range cases {
+		t.Run(name, func(t *testing.T) {
+			pod := &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      map[string]string{},
+					Annotations: map[string]string{},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{Name: "test"}},
+				},
+			}
+			if c.readinessProbe != nil {
+				pod.Spec.Containers[0].ReadinessProbe = c.readinessProbe
+			}
+			if c.livenessProbe != nil {
+				pod.Spec.Containers[0].LivenessProbe = c.livenessProbe
+			}
+
+			h := Handler{
+				EnableTransparentProxy: c.tproxyEnabled,
+				TProxyOverwriteProbes:  c.overwriteProbes,
+			}
+			err := h.overwriteProbes(corev1.Namespace{}, pod)
+			require.NoError(t, err)
+			if c.readinessProbe != nil {
+				require.Equal(t, c.expReadinessPort, pod.Spec.Containers[0].ReadinessProbe.HTTPGet.Port.IntValue())
+				if c.expOriginalReadinessPort != 0 {
+					require.Equal(t, strconv.Itoa(c.expOriginalReadinessPort), pod.Annotations[annotationOriginalReadinessProbePort])
+				}
+			}
+			if c.livenessProbe != nil && c.expLivenessPort != 0 {
```

Comment:
qq: why does the `if` check on L1404 look different (not symmetrical) to the one on L1410? They seems equivalent right? am i missing something?

Reply:
Turned out it wasn't need and was probably left over from another iteration of this test. Thanks so much for catching this 😄